SystemMicrocontroller_Motors/Core/Src/L298N_Motor_driver.c

```diff
@@ -116,22 +137,37 @@ void L298N_Get_Default_Config(L298N_HandleTypeDef * L298N){
 	L298N->DirectionPin_2_Port = GPIOA;
 }
 
-void L298N_Set_Direction(L298N_HandleTypeDef *L298N, uint32_t Direction){
+/**
+ * Function: L298N_Set_Direction
+ * @brief Set direction FORWARD or REVERSE in the L298N device
+ *
+ * @param L298N L298N - The L298N_HandleTypeDef handle
+ * @param Direction IDLE    = 0;
+ * 					FORWARD = 1;
+ * 					REVERSE = 2;
+ */
+void L298N_Set_Direction(L298N_HandleTypeDef *L298N, L298N_Direction Direction){
 	HAL_GPIO_WritePin(L298N->DirectionPin_1_Port, L298N->DirectionPin_1, RESET);
 	HAL_GPIO_WritePin(L298N->DirectionPin_2_Port, L298N->DirectionPin_2, RESET);
 
-	if ( FORWARD == Direction) {
+	if ( L298N_FORWARD == Direction) {
 		HAL_GPIO_WritePin(L298N->DirectionPin_1_Port, L298N->DirectionPin_1, SET);
 		HAL_GPIO_WritePin(L298N->DirectionPin_2_Port, L298N->DirectionPin_2, RESET);
-	}else if (REVERSE == Direction){
+	}else if (L298N_REVERSE == Direction){
 		HAL_GPIO_WritePin(L298N->DirectionPin_1_Port, L298N->DirectionPin_1, RESET);
 		HAL_GPIO_WritePin(L298N->DirectionPin_2_Port, L298N->DirectionPin_2, SET);
 	} else {
 		/**<Do nothing */
 	}
```

Comment:
Perhaps add a third check for if( L298N_IDLE == Direction ) and else throw an 'unexpected input' error? As it currently is, any value of Direction other than 1 or 2 would idle the motor